priam/src/main/java/com/netflix/priam/identity/token/TokenRetrieverUtils.java

```diff
@@ -0,0 +1,122 @@
+package com.netflix.priam.identity.token;
+
+import com.netflix.priam.identity.PriamInstance;
+import com.netflix.priam.utils.SystemUtils;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Common utilities for token retrieval. */
+public class TokenRetrieverUtils {
+    private static final Logger logger = LoggerFactory.getLogger(TokenRetrieverBase.class);
+    private static final String GOSSIP_INFO_URL_FORMAT =
+            "http://%s:8080/Priam/REST/v1/cassadmin/gossipinfo";
+
+    /**
+     * Utility method to infer the IP of the owner of a token in a given datacenter. This method
+     * uses Cassandra gossip information to find the owner. The IP with maximum votes is inferred to
+     * be the token owner.
+     *
+     * @param allIds
```

Comment:
Javadoc please.

Reply:
Updated the Javadoc to reflect the exact behavior.